db/cache/charts.go

```diff
@@ -958,58 +985,94 @@ func accumulate(data ChartUints) ChartUints {
 	return d
 }
 
-// Translate the times slice to a slice of differences. The original dataset
-// minus the first element is returned for convenience.
-func blockTimes(blocks ChartUints) (ChartUints, ChartUints) {
-	times := make(ChartUints, 0, len(blocks))
+// blockTimes enables two line graph charts showing the expected and actual
+// blocks count mined by a given time duration to be generated.
+// It translates the times slice to a slice of differences. The original dataset
+// minus the first element is returned for convenience. limitBlocks defines the
+// number of blocks from the best block (most recent block mined) back towards
+// the genesis block that were supposed to be mined in the limit string passed as
+// the API limit parameter. When the limit string API parameter selected is 'all'
+// the limitBlocks value passed is 0 indicating that all blocks will be used.
+func blockTimes(blocks ChartUints, limitBlocks int) (ChartUints, ChartUints, ChartFloats) {
```

Comment:
Everything for the theoretical distribution should be done client side. No changes needed here.  The only values you need to plot the distribution are the target block time, which the charts controller already has, and the number of blocks, which is already encoded in your data. 

As a general rule, if you can avoid sending data by pushing the processing to JS, you should.